pysidt/sidt.py

```diff
@@ -310,26 +311,101 @@ def evaluate(self, mol):
         return node.rule
 
 
+def to_dict(obj):
+    out_dict = dict()
+    out_dict["class"] = obj.__class__.__name__
+    attrs = [attr for attr in dir(obj) if not attr.startswith("_")]
+    for attr in attrs:
+        val = getattr(obj, attr)
+
+        if callable(val) or val == getattr(obj.__class__, attr):
+            continue
+
+        try:
+            json.dumps(val)
+            out_dict[attr] = val
+        except:
+            if isinstance(val, ScalarQuantity):
+                d = {
+                    "class": val.__class__.__name__,
+                    "value": val.value,
+                    "units": val.units,
+                    "uncertainty": to_dict(val.uncertainty),
+                    "uncertainty_type": val.uncertainty_type,
+                }
+            else:
+                out_dict[attr] = to_dict(val)
+
+    return out_dict
+
+
+def from_dict(d, class_dict=None):
```

Comment:
Can you either rename `d` to be something more informative or add a docstring for this function to explain what is the difference between `class_dict` and `d`

Reply:
add docstring